File: 프로그래머스/2/12911. 다음 큰 숫자/다음 큰 숫자.py
def solution(n):
    answer = 0
    num = n + 1
    binary = bin(n)[2:]
    cnt = 0
    for i in range(len(binary)):
        if binary[i] == '1':
            cnt += 1
    while True:
        target = bin(num)[2:]
        one = 0
        for i in range(len(target)):
            if target[i] == '1':
                one += 1
                
        if one == cnt:
            answer = num
            break
            
        num += 1
        
    return answer

File: 프로그래머스/2/12911. 다음 큰 숫자/다음 큰 숫자.java
class Solution {
    public int solution(int n) {
        int answer = n+1;
        int count = Integer.bitCount(n);
        
        while (true) {
            if (count == Integer.bitCount(answer)) {
                break;
            }
            answer++;
        }
        return answer;
    }
}